src/realm/sync/noinst/client_impl_base.cpp

```diff
@@ -1688,8 +1688,10 @@ void Session::activate()
     if (REALM_LIKELY(!get_client().is_dry_run())) {
         bool file_exists = util::File::exists(get_realm_path());
         m_performing_client_reset = get_client_reset_config().has_value();
+        m_fresh_realm_download = client_reset::is_fresh_path(get_realm_path());
 
-        logger.info("client_reset_config = %1, Realm exists = %2 ", m_performing_client_reset, file_exists);
+        logger.info("client_reset_config = %1, Realm exists = %2, fresh realm download = %3",
+                    m_performing_client_reset, file_exists, m_fresh_realm_download ? "yes" : "no");
         if (!m_performing_client_reset) {
```

Comment:
can't we get rid off this check since we overwrite all members after a client reset? You could actually remove the member all together. we'd also don't need the new check when requesting a file ident in bind.

Reply:
I looked into changing this logic and it is now setting these members during `activate()` but I couldn't get rid of `m_performing_client_reset`, since it is needed to know whether to hold of sending the IDENT message and perform the client reset diff once the BIND message has been sent. I did rename that member var to `m_has_client_reset_config` since that is really all it means.

I couldn't use the `get_client_reset_config()` because it includes checks to ensure the session is Active or Unactivated.

Reply:
I am not sure I follow. When the session sends BIND it is already active so you should be able to use `get_client_reset_config()`.

Reply:
I am performing the client reset diff in the `Session::message_sent()` callback so the session waits until the BIND message has been sent successfully. Since `message_sent()` can be called even when the state is not Active, using the logic of `if (m_bind_message_sent && get_client_reset_config())` can cause an exception, such as when the Session is shutting down (e.g. Deactivating state).

Reply:
got it. you can argue that you should not client reset if the session is Deactivating, or just allow retrieving the config in that state. Out of curiosity, do any tests throw that exception?

Reply:
They did, but I don't remember which one it was...
I can keep working on it if you think it's worth updating.

Reply:
It actually started with the "C API - client reset: Manual reset" test, which is the first test run by passing "[client reset]" from the command line.
I looked at it again and removed the member variable.